feat(wordpress-compose): Inter-app communication (#310)
kraftcloud/examples: wordpress-compose/README.md

Reviewer: change name of example from `wordpress-compose` to `wordpress` for example

> cd examples/wordpress-compose/

Reviewer: change name of example from `wordpress-compose` to `wordpress` for example

```

Make sure to log into Unikraft Cloud by setting your token and a [metro](https://unikraft.com/docs/platform/metros) close to you.
Make sure to log into Unikraft Cloud and pick a [metro](https://unikraft.com/docs/platform/metros) close to you.
This guide uses `fra` (Frankfurt, 🇩🇪):

```bash
```bash title="unikraft"
unikraft login
```

or

```bash title="kraft"
# Set Unikraft Cloud access token
export UKC_TOKEN=token
# Set metro to Frankfurt, DE
export UKC_METRO=fra
```

> When done, invoke the following command to deploy the app on Unikraft Cloud:

Reviewer: remove list numbering from here onwards

> unikraft volume create --metro fra --name wordpress-compose-wordpress-data --size 512M

Reviewer: ```suggestion
unikraft volume create --metro fra --name wordpress-wordpress-data --size 512M
```

> unikraft volume create --metro fra --name wordpress-compose-db-data --size 512M

Reviewer: ```suggestion
unikraft volume create --metro fra --name wordpress-db-data --size 512M
```

```

or

```bash title="kraft"

> kraft cloud volume create --name wordpress-compose-wordpress-data --size 512Mi

Reviewer: ```suggestion
kraft cloud volume create --name wordpress-wordpress-data --size 512Mi
```

> kraft cloud volume create --name wordpress-compose-db-data --size 512Mi

Reviewer: ```suggestion
kraft cloud volume create --name wordpress-db-data --size 512Mi
```

```

> 

Reviewer 2: output for volume creation can also be included for being consistent with instance creation section

but it is not necessary because no information is used further on in the example

it is up to you if you want to include them or not

4. Build and deploy the MariaDB instance:

```bash title="unikraft"

> unikraft build mariadb --output <my-org>/mariadb:latest

Reviewer: ```suggestion
unikraft build ./mariadb --output <my-org>/mariadb:latest
```

> unikraft run --name mariadb --scale-to-zero policy=idle,cooldown-time=1000,stateful=true --metro fra -m 1G --volume wordpress-compose-db-data:/var/lib/mysql --image <my-org>/mariadb:latest --domain=wordpress-compose-mariadb.internal --env MARIADB_ROOT_PASSWORD=unikraft

Reviewer: ```suggestion
unikraft run --name mariadb --scale-to-zero policy=idle,cooldown-time=1000,stateful=true --metro fra -m 1G --volume wordpress-db-data:/var/lib/mysql --image <my-org>/mariadb:latest --domain wordpress-mariadb.internal --env MARIADB_ROOT_PASSWORD=unikraft
```

```

or

```bash title="kraft"

> kraft cloud deploy -M 1G --name mariadb --scale-to-zero idle --scale-to-zero-cooldown 1s --scale-to-zero-stateful --volume wordpress-compose-db-data:/var/lib/mysql --domain=wordpress-compose-mariadb.internal mariadb/ --env MARIADB_ROOT_PASSWORD=unikraft

Reviewer: ```suggestion
kraft cloud deploy -M 1Gi --name mariadb --scale-to-zero idle --scale-to-zero-cooldown 1s --scale-to-zero-stateful --volume wordpress-db-data:/var/lib/mysql --domain wordpress-mariadb.internal ./mariadb/ --env MARIADB_ROOT_PASSWORD=unikraft
```

```

The output shows the instance details:

```ansi title="kraft"
[●] Deployed successfully!
 │
 ├───────── name: mariadb
 ├───────── uuid: d7a5903a-0e36-4be5-ad6f-7eaa58f91fe5
 ├──────── metro: https://api.fra.unikraft.cloud/v1
 ├──────── state: starting

>  ├─────── domain: wordpress-compose-mariadb.internal

Reviewer: ```suggestion
 ├─────── domain: wordpress-mariadb.internal
```

 ├──────── image: oci://unikraft.io/<my-org>/mariadb@sha256:6e2b29ad596dfc1cf11015711455220ba5763e8031e8484c53a5fe4f3591da2f

>  ├─────── memory: 953 MiB

Reviewer: ```suggestion
 ├─────── memory: 1024 MiB
```

 ├────── service: bold-shadow-srhsboq2
 ├─ private fqdn: mariadb.internal
 └─── private ip: 10.0.0.45
```

or

```ansi title="unikraft"
metro:         fra
name:          mariadb

> uuid:          897bfc4a-4023-4787-b430-5671d9892fab

Reviewer: ```suggestion
uuid:          d7a5903a-0e36-4be5-ad6f-7eaa58f91fe5
```

state:         starting
image:         <my-org>/mariadb
resources:
  memory:      1GiB
  vcpus:       1
service:
  uuid:        ae256846-ccb0-4a3d-a42f-e75e144485b8

>   name:        snowy-dust-arru8w8j

Reviewer: ```suggestion
  name:        bold-shadow-srhsboq2
```

  domains:

>   - fqdn:      wordpress-compose-mariadb.internal

Reviewer: ```suggestion
  - fqdn:      wordpress-mariadb.internal
```

volumes:

> - wordpress-compose-db-data:/var/lib/mysql

Reviewer: ```suggestion
- wordpress-db-data:/var/lib/mysql
```

networks:
- uuid:        4c84925a-c05a-429a-b5be-bca032b3cc3c
  private-ip:  10.0.0.45
  mac:         12:b0:0a:00:00:2d
timestamps:
  created:     just now
scale-to-zero: policy=idle,stateful=true,cooldown-time=1s
```

5. Build and deploy the Wordpress instance:

```bash title="unikraft"

> unikraft build wordpress --output <my-org>/wordpress:latest

Reviewer: ```suggestion
unikraft build ./wordpress --output <my-org>/wordpress:latest
```

> unikraft run --name wordpress --scale-to-zero policy=on,cooldown-time=1000 --metro fra -p 443:8080/tls+http -m 2G --volume wordpress-compose-wordpress-data:/var/www/html --image <my-org>/wordpress:latest --env WORDPRESS_DB_HOST=wordpress-compose-mariadb.internal

Reviewer: ```suggestion
unikraft run --name wordpress --scale-to-zero policy=on,cooldown-time=1000 --metro fra -p 443:8080/tls+http -m 2G --volume wordpress-wordpress-data:/var/www/html --image <my-org>/wordpress:latest --env WORDPRESS_DB_HOST=wordpress-mariadb.internal
```

```

or

```bash title="kraft"

> kraft cloud deploy -M 2G --name wordpress --scale-to-zero on --scale-to-zero-cooldown 1s --volume wordpress-compose-wordpress-data:/var/www/html -p 443:8080/tls+http wordpress/ --env WORDPRESS_DB_HOST=wordpress-compose-mariadb.internal

Reviewer: ```suggestion
kraft cloud deploy -M 2Gi --name wordpress --scale-to-zero on --scale-to-zero-cooldown 1s --volume wordpress-wordpress-data:/var/www/html -p 443:8080/tls+http ./wordpress/ --env WORDPRESS_DB_HOST=wordpress-mariadb.internal
```

```

The output shows the instance address and other details:

```ansi title="kraft"
[●] Deployed successfully!
 │
 ├───────── name: wordpress
 ├───────── uuid: 3bf59963-4b17-4672-9ba4-b9ac2f877014
 ├──────── metro: https://api.fra.unikraft.cloud/v1
 ├──────── state: starting
 ├─────── domain: https://funky-pine-7p7lqzk5.fra.unikraft.app
 ├──────── image: oci://unikraft.io/<my-org>/wordpress@sha256:d13ace75c8515ac369b8953d53cdefffc1ba5037057a386e2f17252b6041e091

>  ├─────── memory: 1907 MiB

Reviewer: ```suggestion
 ├─────── memory: 2048 MiB
```

 ├────── service: funky-pine-7p7lqzk5
 ├─ private fqdn: wordpress.internal
 └─── private ip: 10.0.0.33
```

or

```ansi title="unikraft"
metro:         fra
name:          wordpress

> uuid:          ee9f5599-b33b-44f1-ab57-3871b440e810

Reviewer: ```suggestion
uuid:          3bf59963-4b17-4672-9ba4-b9ac2f877014
```

state:         starting
image:         <my-org>/wordpress
resources:
  memory:      2GiB
  vcpus:       1
service:
  uuid:        7f21d778-487f-4130-a94a-34b86862c3dd

>   name:        damp-lake-n09gzguc

Reviewer: ```suggestion
  name:        funky-pine-7p7lqzk5
```

  domains:

>   - fqdn:      damp-lake-n09gzguc.fra.unikraft.app

Reviewer: ```suggestion
  - fqdn:      funky-pine-7p7lqzk5.fra.unikraft.app
```

volumes:

> - wordpress-compose-wordpress-data:/var/www/html

Reviewer: ```suggestion
- wordpress-wordpress-data:/var/www/html
```

networks:
- uuid:        e0f13623-fd04-4a69-9caf-47142ce47c4c
  private-ip:  10.0.0.33
  mac:         12:b0:0a:00:00:21
timestamps:
  created:     just now
scale-to-zero: policy=on,cooldown-time=1s
```

Use a browser to access the install page of Wordpress using the URL from the `domain` / `fqdn` field in the output.
Fill out the form and complete the Wordpress install.

> 

Reviewer: Add listing output for the instances:
`You can list information about the instances by running:
...
`

When done, you can remove the instances:

```bash title="unikraft"
unikraft instances delete mariadb

> unikraft instances delete wordpress

Reviewer: ```suggestion
unikraft instances delete mariadb wordpress
```

```

or

```bash title="kraft"
kraft cloud instance remove mariadb

> kraft cloud instance remove wordpress

Reviewer: ```suggestion
kraft cloud instance remove mariadb wordpress
```

```

> 

Reviewer 2: also remove the volumes
